src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -193,4 +204,88 @@ object BoringUtils {
     sinks.foreach(addSink(_, genName, true, true))
     genName
   }
+
+  /** Access a source [[Data]] that may or may not be in the current module.  If
+    * this is in a child module, then create ports to allow access the
+    * requrested source.
+    */
+  def bore(source: Data)(implicit si: SourceInfo): Data = {
+    import reflect.DataMirror
+    def parent(d: Data): experimental.BaseModule = d.topBinding.location.get
+    def purePortType = if (DataMirror.hasOuterFlip(source)) Flipped(chiselTypeOf(source)) else chiselTypeOf(source)
+    def emptyWireOnError = Wire(purePortType) //return empty wire to continue elaboration to collect other errors
+
+    experimental.requireIsHardware(source)
+    val thisModule = Builder.currentModule.get
+    source.topBindingOpt match {
+      case None =>
+        Builder.error(s"Cannot bore from ${source._errorContext}")
+        return emptyWireOnError
+      case Some(internal.CrossModuleBinding) =>
+        Builder.error(
+          s"Cannot bore across a Definition/Instance boundary:${thisModule._errorContext} cannot access ${source}"
+        )
+        return emptyWireOnError
+      case _ => // Actually bore
+    }
+    if (parent(source) == thisModule) {
+
+      /** No boring to do */
+      return source
+    }
+
+    val lca = DataMirror.leastCommonAncestorModule(source, thisModule)
+    if (lca.isEmpty) {
+      Builder.error(s"Cannot bore from $source to ${thisModule.name}, as they do not share a least common ancestor")
+      return Wire(purePortType) //return empty wire to continue elaboration to collect other errors
+    }
+
+    val upPath = DataMirror.modulePath(source, lca)
+    val downPath = DataMirror.modulePath(thisModule, lca)
```

Comment:
This organization is a bit odd. The LCA method is used to find the LCA using `modulePath`. Then `modulePath` is re-used to find the path. Would a single method like: `findLCAPaths(foo, bar): (Seq[BaseModule], Seq[BaseModule]) = ???` work better?